apps/sms/services/js/threads.js

```diff
@@ -6,16 +6,24 @@
   var threads = new Map();
   var messageMap = new Map();
 
-  function Thread(thread) {
-    var length = Thread.FIELDS.length;
-    var key;
-
-    for (var i = 0; i < length; i++) {
-      key = Thread.FIELDS[i];
-      this[key] = thread[key];
-    }
+  function extractSupportedFields(
+    { id, body, timestamp, participants, lastMessageType, status, isDraft }
+  ) {
```

Comment:
nit: i'm not sure of this, I don't find this so readable; I'd rather keep Thread.FIELDS... (but you can keep the function though)